### ISSUE TITLE
Question on error "DMD executable path is not specified correctly"

### ISSUE THREAD

**Issue by DLDavid** [intellij-dlanguage/intellij-dlanguage#398]
Using Intellij 2017.3 with D Plug in 1.16.1 and looking for an answer to configuration problem
"DMD executable path is not specified correctly"
Error running 'UnitTest': Cannot run program "/usr/local/Cellar/dmd" (in directory "/Users/me/IdeaProjects/UnitTestEx"): error=13, Permission denied"
>>
 I can run unittest on the console but get the above error when in the IDE. Can you provide what causes this issue?

**Comment by pirocks** [comment]
Is `/usr/local/Cellar/dmd` a dmd executable? Can you run it/does your user
account have permission to access it?
You can also try running type -p dmd which will tell you the location of
dmd executable you use currently.

**Comment by DLDavid** [comment]
This is base d compiler directory "2.071.1". This base and subsequent subdirectories I have have full privileges. Do I need to included the full path to bin directory?

**Comment by pirocks** [comment]
> Do I need to included the full path to bin directory?

In short, yes. I am looking into finding a way of making this more clear during the setup process/ improving setup in general.

Here's a screenshot of what a correctly setup sdk should look like under File> Project Structure > SDKs

![image](https://user-images.githubusercontent.com/10455020/36865677-1a6245f4-1d88-11e8-9bf7-44c45d833ea2.png)

**Comment by SingingBush** [comment]
on mac it should look for dmd installed via homebrew by default. `/usr/local/Cellar/dmd` looks wrong. been a while since I was on mac but pretty sure that homebrew makes symlinks for everything that gets installed.  Therefore we check for the full binary path: `/usr/local/opt/dmd`

on windows it would make sense to look for the base directory of the dmd installation but on Linux/Mac that's not the case. see #227

**Comment by DLDavid** [comment]
Yes, it worked after fully qualifying the path. one additional question is it possible to add other options to the compiler other then what is predefined in the dialog box?
Thanks.
![screen shot 2018-03-01 at 8 38 03 pm](https://user-images.githubusercontent.com/13157351/36880933-fcfbf1e6-1d90-11e8-9673-ef1171d8e2e4.png)

**Comment by pirocks** [comment]
![image](https://user-images.githubusercontent.com/10455020/36889942-01544000-1df4-11e8-92a0-85d450dd8025.png)

When compiling with dmd you can use the checkboxes/option shown in the screenshot above. You can also manually edit arguments under the Arguments tab. You can get to this menu via "Edit Configurations" as shown below:
![image](https://user-images.githubusercontent.com/10455020/36890045-64ecedf6-1df4-11e8-9a36-ba7230ac7db5.png)

**Comment by DLDavid** [comment]
I went to the Argument page and typed in the **Actual DMD Compiler arguments:** window. I clicked apply (or OK) but the changes were NOT accepted. Is there something I need  to have it accepted or is there a method to enable editing the window? This is on a macOS Sierra, Intellij 2017.3, D plugin 1.16.1.

**Comment by SingingBush** [comment]
> You can also manually edit arguments under the Arguments tab

The arguments tab just shows what is being passed to dmd. It's not directly editable. Changing the options on the other 3 tabs (Compilation, Output, Debug) will affect the result of the args that are displayed in the Arguments tab.

I just tried this with a few options and clicking apply. When I ran the dmd task the compiler args were being used as intended

**Comment by DLDavid** [comment]
So if the command line option is not on one of the 3 tabs is there a way to add one of the other switches? 
Are some options not available from the configuration tabs like -main  or -betterC for example? Is there another way to set this options?

**Comment by pirocks** [comment]
>  Is there another way to set this options?

As far as I know no. @SingingBush might know more.

**Comment by SingingBush** [comment]
no, we just need to add the missing compiler options to that screen. Or just allow users to type args into a text field. If we add -betterC, perhaps there should be a check to ensure that the dmd version is recent enough to support it. Seeing as there's a ton of compiler options and it's a moving target, we're probably better off letting the user type in whatever args they want.
